src/test/java/com/list/todo/it/TagTest.java

```diff
@@ -0,0 +1,510 @@
+                                      NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
+            return new PageRequest(0, 50);
+        }
+    };
+
+    @Before
+    public void before() {
+        mockMvc = MockMvcBuilders
+                .standaloneSetup(tagControllerMock)
+                .setCustomArgumentResolvers(putAuthenticationPrincipal, putSupportParameters)
+                .build();
+    }
+
+    @Test
+    public void getMyTags_ReturnsAListOfTags() throws Exception {
+        //arrange
+        when(tagServiceMock.getTagsByOwnerId(CURRENT_USER_ID)).thenReturn(getListOfTags());
+
+        //act
+        MvcResult result = this.mockMvc.perform(get("/api/tags"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn();
+        //assert
+        assertEquals(getTagsFromJsonResponse(result.getResponse().getContentAsString()), getListOfTags());
+
+        verify(tagServiceMock, times(1)).getTagsByOwnerId(CURRENT_USER_ID);
+
+    }
+
+    @Test
+    public void getMyTagsWithTodoListId_OnExistentTagsIds_ReturnsAListOfTagTaskKeys() throws Exception {
+        //arrange
+        List<Long> tagsIds = new ArrayList<Long>() {{
+            add(6L);
+        }};
+
+        when(tagTaskKeyServiceMock.getMyTaggedTask(any(UserPrincipal.class), any(Pageable.class), eq(tagsIds)))
+                .thenReturn(getListOfTaggedTask());
+
+        //act
+        MvcResult result = this.mockMvc.perform(get("/api/tags/myTagTaskKeys?tagId=6"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn();
+
+        //assert
+        assertEquals(new HashSet<>(getTagTaskKeyFromJsonResponse(result.getResponse().getContentAsString())), getListOfTaggedTask());
+
+        verify(tagTaskKeyServiceMock, times(1)).getMyTaggedTask(any(UserPrincipal.class), any(Pageable.class), eq(tagsIds));
+    }
+
+    @Test
+    public void addTag_ReturnsAObjectOfTag() throws Exception {
+        //arrange
+        TagInput tagInput = new TagInput("Home", "ff");
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        when(tagServiceMock.addTag(tagInput, CURRENT_USER_ID)).thenReturn(Optional.of(new Tag("Home", CURRENT_USER_ID, "ff")));
+
+        //act, assert
+        this.mockMvc.perform(post("/api/tags")
+                .content(objectMapper.writeValueAsString(tagInput))
+                .contentType(MediaType.APPLICATION_JSON_UTF8))
+                .andDo(print())
+                .andExpect(jsonPath("tagName").value(tagInput.getTagName()))
+                .andExpect(jsonPath("ownerId").value(CURRENT_USER_ID))
+                .andExpect(status().isOk());
+
+        verify(tagServiceMock, times(1)).addTag(tagInput, CURRENT_USER_ID);
+    }
+
+    @Test
+    public void updateTag_OnExistentTag_ReturnsAObjectOfTag() throws Exception {
+        //arrange
+        Tag tag = new Tag("Home", CURRENT_USER_ID, "ff");
+        tag.setId(2L);
```

Comment:
Don't use "magic numbers"

Reply:
Especially if it is some meaningful value. Especially if it is used in code several times